src/Orleans.Core/Configuration/CollectionAgeLimitAttribute.cs

```diff
@@ -8,6 +8,25 @@ namespace Orleans.Configuration
     [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
     public class CollectionAgeLimitAttribute : Attribute
     {
-        public double Minutes { get; set; }
+        public const int DefaultAgeMinutes = 10;
+
+        public int Days { get; set; } = 0;
+        public int Hours { get; set; } = 0;
+        public int Minutes { get; set; } = 0;
+
+        public bool AlwaysActive { get; set; }
+
+        public TimeSpan Amount
+        {
+            get
+            {
+                var span = AlwaysActive
+                ? TimeSpan.FromDays(short.MaxValue)
+                : TimeSpan.FromDays(Days) + TimeSpan.FromHours(Hours) + TimeSpan.FromMinutes(Minutes);
+                return span < TimeSpan.FromMinutes(DefaultAgeMinutes)
```

Comment:
Why this comparison? Would we want to return the default unless any of the values - `Days`, `Hours`, `Minutes` - is non-zero?

Reply:
I think the ```DefaultAgeMinutes``` should recommend value , should I change this line to ```span < TimeSpan.Zero``` ?

> Why this comparison? Would we want to return the default unless any of the values - Days, Hours, Minutes - is non-zero?

Reply:
The logic I think is right is in https://github.com/bluexo/orleans/pull/1.